### PR TITLE
feat: add documentation for windows-customize and windows-efi-installer example pipelines

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
feat: add documentation for windows-customize and windows-efi-installer example pipelines
This commits adds/updates documentation for windows-customize and windows-efi-installer example pipelines. These pipelines can be used to install and customize win10, 11, 2k22 from iso with tekton.

Signed-off-by: Karel Šimon <ksimon@redhat.com>

**Release note**:
```
Add documentation for windows-customize and windows-efi-installer example pipelines
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,10 @@ Visit [RBAC permissions for running the tasks](docs/tasks-rbac-permissions.md) i
 
 - [wait-for-vmi-status](tasks/wait-for-vmi-status)
 
+#### Modify Windows iso
+- [modify-windows-iso-file](tasks/modify-windows-iso-file) - modifies windows iso (replaces prompt bootloader with no-promt
```
Comment (0xFelix) [inline README.md:74]: Can you explain why one should use the noprompt bootloader?

File: examples/pipelines/windows-customize/README.md [hunk 2]
```diff
@@ -1,11 +1,15 @@
-# Windows 10 Customize Pipeline
+# Windows Customize Pipeline
 
-This pipeline clones the DataVolume of a basic and generalized Windows 10 installation and runs arbitrary customization
+This pipeline clones the DataVolume of a basic and generalized Windows 10/11/2k22 installation and runs arbitrary customization
 commands through an unattend.xml after startup  of the VM. As an example a ConfigMap which installs Microsoft SQL
-Server Express and generalizes the VM after is included (`windows10-sqlserver`). For basic setup after the first start
-of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`.
+Server Express and generalizes the VM after is included (e.g. `windows10-sqlserver`, or `windows11-vs-code`). For basic setup after the first start
```
Comment (0xFelix) [inline examples/pipelines/windows-customize/README.md:5]: `...or a ConfigMap that install VSCode. (windows11-vs-code)`?

File: examples/pipelines/windows-customize/README.md [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`, or `windows11-vs-code`.
 
-The provided reference ConfigMap (`windows10-sqlserver`) boots Windows 10 into Audit mode, applies the customizations as
+This example pipeline can be used for running windows 10, 11 or 2k22 (or others - not tested!). Always adjust pipeline parameters 
+for windows version you are currently using(e.g. different template name, different base image name, ...). It is possible to use 
+`windows10-unattend` config map for win11/2k22 and vice versa (`windows11-vs-code` for win10/2k22).
```
Comment (0xFelix) [inline examples/pipelines/windows-customize/README.md:10]: Is it worth mentioning this or why is there a need for different config maps at all then?

File: examples/pipelines/windows-efi-installer/README.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+This example pipeline is suitable only for windows 11/2k22 (or other windows versions which require efi - not tested!). When using 
+this example pipeline always adjust pipeline parameters for windows version you are currently using(e.g. different template name, 
+different autoattend config map, different base image name, ...). Each windows version requires change in autounattendConfigMapName 
+parameter (e.g. using `windows2k22-autounattend` config map will not work with Windows 11 and vice versa). When creating 
+pipelineRun, always specify `spec.taskRunSpecs[*].taskPodTemplate.securityContext` for `pipelineTaskName: 
+"modify-windows-iso-file"` as stated in [example](windows11-installer-pipelinerun-okd.yaml), otherwise `modify-windows-iso-file` 
+task will fail
+
+The pipeline implements this by modifying the windows iso (extracts iso files from iso, replaces prompt bootloader with no-prompt bootloader, pack updated packages to new iso, converts it with qemu-img and replaces original iso file in PVC), spinning up a new VM which boots from the Windows installation image (ISO file). The
```
Comment (0xFelix) [inline examples/pipelines/windows-efi-installer/README.md:14]: Can you explain the need for the nopromt bootloader?

File: examples/pipelines/windows-efi-installer/README.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+   Make sure to escape `&` with `\&` for the example commands below to work.
+
+```bash
+# Real URL can look differently
+WIN_URL="https://software.download.prss.microsoft.com/dbazure/Win11_22H2_English_x64v1.iso..."
+```
+
+#### Obtaining a download URL in an automated way
+
+The script [`getisourl.py`](getisourl.py) can be used to automatically obtain a Windows 10 ISO download URL.
```
Comment (0xFelix) [inline examples/pipelines/windows-efi-installer/README.md:49]: Does this apply Windows 2k22 too? Update text to matching Windows version.

File: examples/pipelines/windows-efi-installer/README.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+1. Supply, generate or use the default autounattend.xml.
+   For information on answer files see [Startup Scripts - KubeVirt User Guide](https://kubevirt.io/user-guide/virtual_machines/startup_scripts/#sysprep).
+2. Replace the default example autounattend.xml with your own in the definition of the `windows11-autounattend` ConfigMap in the pipeline YAML.
+   Different autounattend.xml can be also passed in a separate ConfigMap with the Pipeline parameter `autounattendConfigMapName` when creating a PipelineRun.
+
+## Pipeline Description (Kubernetes)
+
+```
+  create-source-dv --- create-vm-from-manifest --- wait-for-vmi-status --- cleanup-vm
```
Comment (0xFelix) [inline examples/pipelines/windows-efi-installer/README.md:74]: No modify iso task?

File: README.md (hunk 1 above)
Comment (ksimon1) [reply README.md:74]: That is explained in readme of the task https://github.com/kubevirt/kubevirt-tekton-tasks/pull/219/files#diff-8cb4772fdd977a741135b0b25c759fd92903ecf1121237e85f4a5e345ea95922

File: examples/pipelines/windows-customize/README.md [hunk 7]
```diff
@@ -1,11 +1,16 @@
...
+# Windows Customize Pipeline
 
-This pipeline clones the DataVolume of a basic and generalized Windows 10 installation and runs arbitrary customization
-commands through an unattend.xml after startup  of the VM. As an example a ConfigMap which installs Microsoft SQL
-Server Express and generalizes the VM after is included (`windows10-sqlserver`). For basic setup after the first start
-of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`.
+This pipeline clones the DataVolume of a basic and generalized Windows 10/11/2k22 installation and runs arbitrary customization
+commands through an unattend.xml after startup of the VM. As an example a ConfigMap which installs Microsoft SQL
+Server Express and generalizes the VM after is included (`windows-sqlserver`) or a ConfigMap that install VSCode (`windows-vs-code`). 
+For basic setup after the first start of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`,
```
Comment (0xFelix) [inline examples/pipelines/windows-customize/README.md:6]: Newline? Also might make sense to drop the `windows10-unattend` and `windows11-unattend` ConfigMaps from TTO and move them here, as they are only used as examples?

File: examples/pipelines/windows-customize/README.md [hunk 8]
```diff
@@ -1,11 +1,16 @@
...
-of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`.
+This pipeline clones the DataVolume of a basic and generalized Windows 10/11/2k22 installation and runs arbitrary customization
+commands through an unattend.xml after startup of the VM. As an example a ConfigMap which installs Microsoft SQL
+Server Express and generalizes the VM after is included (`windows-sqlserver`) or a ConfigMap that install VSCode (`windows-vs-code`). 
+For basic setup after the first start of a customized VM an example unattend.xml is included in the pipeline's ConfigMap `windows10-unattend`, 
+or `windows11-unattend`.
 
-The provided reference ConfigMap (`windows10-sqlserver`) boots Windows 10 into Audit mode, applies the customizations as
+This example pipeline can be used for running windows 10, 11 or 2k22 (or others - not tested!). Always adjust pipeline parameters 
+for windows version you are currently using(e.g. different template name, different base image name, ...). It is possible to use
```
Comment (0xFelix) [inline examples/pipelines/windows-customize/README.md:10]: Insert space before `(`

**Pull request merged**